src/modules/git_status.rs

```diff
@@ -214,6 +214,12 @@ fn get_repo_status(
     log::debug!("New repo status created");
 
     let mut repo_status = RepoStatus::default();
+
+    // can't run git status against a bare repo
+    if repo.repo.work_tree.is_none() {
+        log::debug!("This is a bare repository, not generating git status");
+        return Some(repo_status);
+    }
```

Comment:
This should probably check if the repository kind is `Bare` per the git config. Since `git status` ignores bare repositories in general, perhaps also perform this check earlier.

Reply:
Makes sense, code updated to enable checking the repo type and I've moved the check to the start of the `module` function